libs/datamodel/connectors/sql-datamodel-connector/src/cockroach_datamodel_connector.rs

```diff
@@ -0,0 +1,301 @@
+            DATE_TYPE_NAME => Date,
+            TIME_TYPE_NAME => Time(parse_one_opt_u32(args, TIME_TYPE_NAME)?),
+            TIME_TZ_TYPE_NAME => Timetz(parse_one_opt_u32(args, TIME_TZ_TYPE_NAME)?),
+            BOOLEAN_TYPE_NAME => Boolean,
+            BIT_TYPE_NAME => Bit(parse_one_opt_u32(args, BIT_TYPE_NAME)?),
+            VAR_BIT_TYPE_NAME => VarBit(parse_one_opt_u32(args, VAR_BIT_TYPE_NAME)?),
+            UUID_TYPE_NAME => Uuid,
+            JSON_TYPE_NAME => Json,
+            JSON_B_TYPE_NAME => JsonB,
+            _ => return Err(ConnectorError::new_native_type_parser_error(name)),
+        };
+
+        Ok(NativeTypeInstance::new(name, cloned_args, &native_type))
+    }
+
+    fn introspect_native_type(&self, native_type: serde_json::Value) -> Result<NativeTypeInstance, ConnectorError> {
+        let native_type: CockroachType = serde_json::from_value(native_type).unwrap();
+        let (constructor_name, args) = match native_type {
+            SmallInt => (SMALL_INT_TYPE_NAME, vec![]),
+            Integer => (INTEGER_TYPE_NAME, vec![]),
+            BigInt => (BIG_INT_TYPE_NAME, vec![]),
+            Decimal(x) => (DECIMAL_TYPE_NAME, args_vec_from_opt(x)),
+            Real => (REAL_TYPE_NAME, vec![]),
+            DoublePrecision => (DOUBLE_PRECISION_TYPE_NAME, vec![]),
+            VarChar(x) => (VARCHAR_TYPE_NAME, arg_vec_from_opt(x)),
+            Char(x) => (CHAR_TYPE_NAME, arg_vec_from_opt(x)),
+            Text => (TEXT_TYPE_NAME, vec![]),
+            ByteA => (BYTE_A_TYPE_NAME, vec![]),
+            Timestamp(x) => (TIMESTAMP_TYPE_NAME, arg_vec_from_opt(x)),
+            Timestamptz(x) => (TIMESTAMP_TZ_TYPE_NAME, arg_vec_from_opt(x)),
+            Date => (DATE_TYPE_NAME, vec![]),
+            Time(x) => (TIME_TYPE_NAME, arg_vec_from_opt(x)),
+            Timetz(x) => (TIME_TZ_TYPE_NAME, arg_vec_from_opt(x)),
+            Boolean => (BOOLEAN_TYPE_NAME, vec![]),
+            Bit(x) => (BIT_TYPE_NAME, arg_vec_from_opt(x)),
+            VarBit(x) => (VAR_BIT_TYPE_NAME, arg_vec_from_opt(x)),
+            Uuid => (UUID_TYPE_NAME, vec![]),
+            Json => (JSON_TYPE_NAME, vec![]),
+            JsonB => (JSON_B_TYPE_NAME, vec![]),
+            Inet => (INET_TYPE_NAME, vec![]),
+            Citext => (CITEXT_TYPE_NAME, vec![]),
+            Oid => (OID_TYPE_NAME, vec![]),
+        };
+
+        if let Some(constructor) = self.find_native_type_constructor(constructor_name) {
+            Ok(NativeTypeInstance::new(constructor.name, args, &native_type))
+        } else {
+            Err(self.native_str_error(constructor_name).native_type_name_unknown())
+        }
+    }
+
+    fn validate_url(&self, url: &str) -> Result<(), String> {
+        if !url.starts_with("postgres://") && !url.starts_with("postgresql://") {
+            return Err("must start with the protocol `postgresql://` or `postgres://`.".to_owned());
+        }
```

Comment:
I'm wondering whether we should think about `cockroachdb://` URLs, since we're the ones defining what we accept. It's not immediately relevant in this PR though.

Reply:
Ideally you'd be able to just copy the conn string as is from a source without having to switch it to cockroachdb://, and their online offering provides them with postgres://. But currently you need to modify it anyways it seems.

Reply:
Further internal discussion: https://prisma-company.slack.com/archives/C02K246T9RQ/p1638883489006500

Reply:
Just checking. From the discussion it seems I don't need to change anything here?